### ISSUE TITLE
sample sheet and local files

### ISSUE THREAD

**Issue by mark-lubberts** [KosinskiLab/AlphaPulldownSnakemake#26]
Hi, 

I'm looking for some clarity around how to specify local files in the sample sheet. The README says to use
```
protein:N:start-stop[_protein:N:start-stop]*
```
>where protein is a path to a file with '.fasta' extension or uniprot ID, N is the number of monomers for this particular protein and start and stop are the residues that should be predicted. However, only protein is required, N, start and stop can be omitted.

1. For specifying multiple local files, do I need to create a separate fasta for each protein sequence?
2. The [InputParser](https://github.com/KosinskiLab/AlphaPulldownSnakemake/blob/1fa4efe9a563e13f59a10bf8fa525445f0c99a87/workflow/rules/common.smk#L48) seems to split all underscores, so any underscores in the path to a local file cause the pipeline to try to download various path segments. E.g.
   ```
   /path_to/some_protein/dataset.fasta
   ```
gets treated as if I'm trying to model proteins `path.fasta`, `to.fasta`, `some.fasta`, `protein.fasta`, and `dataset.fasta` together. I tried quoting the filepath, but it didnt' make a difference. Is there a way to get around this?

**Comment by jkosinski** [comment]
Hi Mark,
@DimaMolod is away at the moment, he will check your case, and maybe indeed it's abug, but in the meantime would it be possible to use a UniProt ID instead of the fasta file? In that case, AlphaPulldown will download the sequences automatically. Also, note the more extensive README at the main AlphaPulldown page: https://github.com/KosinskiLab/AlphaPulldown

**Comment by DimaMolod** [comment]
Hi @mark-lubberts 
if you have underscores in fasta filename paths, just use another protein delimiter. You can change it here:
https://github.com/KosinskiLab/AlphaPulldownSnakemake/blob/main/config/config.yaml#L3
You could use a plus symbol ('+') that is rarely used in the file paths.

**Comment by bifxcore** [comment]
This question was not answered: do I need to create a separate fasta for each protein sequence?

From the examples, it seems that the answer is : yes. When the main purpose of this package is to do a pulldown, I would have expected one file for the bait and a single file with all proteome sequences for the pulldown, rather than several thousand files.

**Comment by jkosinski** [comment]
You’re right; with this syntax, using a combined fasta file is not possible; we need to think about how to enable it.

If you don’t want to split the FASTA file, for now, you could follow one of these two options:
1) use UniProt IDs as protein names instead of FASTA files if possible. In that case, AP will download the sequences automatically
2) generate features using the non-snakemake AlphaPulldown interface, which does allow for multi-sequence files https://github.com/KosinskiLab/AlphaPulldown/tree/main?tab=readme-ov-file#1-compute-multiple-sequence-alignment-msa-and-template-features-cpu-stage and then use the features using:
feature_directory :
  - "/path/to/directory/with/features/"
in the snakemake pipeline.

Would any of these work for you?
And thanks for spotting this!

**Comment by bifxcore** [comment]
My sequences are not in UniProt, so option 1 is unfortunately not an option.
Option 2 sounds like the way to go, I'm installing the python command-line interface of AlphaPulldown.
Thanks for the tip!

**Comment by jkosinski** [comment]
OK, great, I will close this issue here then and open an enhancement feature request at https://github.com/KosinskiLab/AlphaPulldown
